extensions/amp-imgur/0.1/amp-imgur.js

```diff
@@ -56,38 +58,69 @@ export class AmpImgur extends AMP.BaseElement {
 
   /** @override */
   buildCallback() {
-    this.imgurid_ = userAssert(
-      this.element.getAttribute('data-imgur-id'),
+    userAssert(
+      this.element.dataset.imgurId,
       'The data-imgur-id attribute is required for <amp-imgur> %s',
       this.element
     );
   }
 
   /** @override */
   layoutCallback() {
-    const iframe = this.element.ownerDocument.createElement('iframe');
-    this.iframe_ = iframe;
-
     this.unlistenMessage_ = listen(
       this.win,
       'message',
       this.handleImgurMessages_.bind(this)
     );
 
-    iframe.setAttribute('scrolling', 'no');
-    iframe.setAttribute('frameborder', '0');
-    iframe.setAttribute('allowfullscreen', 'true');
-
-    const sanitizedID = this.imgurid_.replace(
+    const sanitizedId = this.element.dataset.imgurId.replace(
       /^(a\/)?(.*)/,
-      (match, aSlash, rest) => {
-        return 'a/' + encodeURIComponent(rest);
+      (unusedMatch, aSlash, rest) => (aSlash || '') + encodeURIComponent(rest)
+    );
+
+    return this.insertLoadIframe_(sanitizedId).then(() => {
+      // Unfortunately, between April 2020 and May 2021 we incorrectly interpreted
+      // image ids as album ids, so we'd add an a/ prefix even when unnecessary.
+      // When 404ing, we won't receive messages, so we retry with a/
+      // https://go.amp.dev/issue/28049
```

Comment:
Since this reversal is legitimate, I didn't want to break albums that did not include `a/`, so I'm including a best effort here.

(I can write tests once we agree with the approach.)